src/Symfony/Component/DependencyInjection/ContainerAwareTrait.php

```diff
@@ -19,19 +19,15 @@
 trait ContainerAwareTrait
 {
     /**
-     * @var ContainerInterface
+     * @var ContainerInterface|null
      */
     protected $container;
 
     /**
      * @return void
      */
-    public function setContainer(ContainerInterface $container = null)
+    public function setContainer(?ContainerInterface $container)
     {
-        if (1 > \func_num_args()) {
-            trigger_deprecation('symfony/dependency-injection', '6.2', 'Calling "%s::%s()" without any arguments is deprecated, pass null explicitly instead.', __CLASS__, __FUNCTION__);
```

Comment:
what about deprecating both the trait and the interface?

Reply:
What would be the recommended migration path? "The ContainerAwareTrait/Interface is deprecated, implement the `setContainer()` method with the `#[Required]` attribute instead."?

Reply:
"Use dependency injection instead"

Reply:
..with a nice before/after example in the upgrade guide 🙂

Reply:
Deprecating both in https://github.com/symfony/symfony/pull/50615. Finishing the work on this one, thanks both for your help!